File: workInfo/plugin/MethodInfo.java
package com.example.demo.pojo.plugin;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

/**
 * 方法信息
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class MethodInfo {

    /**
     * 方法id
     * 系统生成系统与用户均不可修改
     */
    private String id;

    /**
     * 方法描述
     * 系统生成默认为空用户可修改
     */
    private String description;

    /**
     * 方法所属方法类ID
     * 系统生成系统与用户均不可修改
     */
    private String methodClassId;

    /**
     * 方法名
     * 系统生成系统与用户均不可修改
     */
    private String name;

    /**
     * 方法参数列表（JSON格式）
     */
    private List<ParameterInfo> parameters;

    /**
     * 方法返回值类型
     * 系统生成系统与用户均不可修改
     */
    private String returnType;
}

File: src/models/MethodInfo.js
/**
 * 方法信息模型
 */
import { ParameterInfo } from './ParameterInfo';

export class MethodInfo {
  /**
   * 方法id
   * 系统生成系统与用户均不可修改
   */
  id = '';

  /**
   * 方法描述
   * 系统生成默认为空用户可修改
   */
  description = '';

  /**
   * 方法所属方法类ID
   * 系统生成系统与用户均不可修改
   */
  methodClassId = '';

  /**
   * 方法名
   * 系统生成系统与用户均不可修改
   */
  name = '';

  /**
   * 方法参数列表
   */
  parameters = [];

  /**
   * 方法返回值类型
   * 系统生成系统与用户均不可修改
   */
  returnType = '';

  /**
   * 方法返回值描述
   * 系统生成默认为空用户可修改
   */
  returnDescription = '';

  constructor(data = {}) {
    Object.assign(this, data);
    if (data.parameters) {
      this.parameters = data.parameters.map(param => new ParameterInfo(param));
    }
  }
}